packages/protocol/contracts/governance/EpochRewards.sol

```diff
@@ -0,0 +1,399 @@
+      max,
+      underspendAdjustmentFactor,
+      overspendAdjustmentFactor
+    );
+    return true;
+  }
+
+  /**
+   * @notice Sets the target voting yield parameters.
+   * @param max The max relative target block reward for voters.
+   * @param adjustmentFactor The target block reward adjustment factor for voters.
+   * @return True upon success.
+   */
+  function setTargetVotingYieldParameters(
+    uint256 max,
+    uint256 adjustmentFactor
+  )
+    public
+    onlyOwner
+    returns (bool)
+  {
+    require(
+      max != targetVotingYieldParams.max.unwrap() ||
+      adjustmentFactor != targetVotingYieldParams.adjustmentFactor.unwrap()
+    );
+    targetVotingYieldParams.max = FixidityLib.wrap(max);
+    targetVotingYieldParams.adjustmentFactor = FixidityLib.wrap(adjustmentFactor);
+    require(
+      targetVotingYieldParams.max.lt(FixidityLib.fixed1()),
+      "Max target voting yield must be lower than 100%"
+    );
+    emit TargetVotingYieldParametersSet(max, adjustmentFactor);
+    return true;
+  }
+
+  /**
+   * @notice Returns the target gold supply according to the epoch rewards target schedule.
+   * @return The target gold supply according to the epoch rewards target schedule.
+   */
+  function getTargetGoldTotalSupply() public view returns (uint256) {
+    uint256 timeSinceInitialization = now.sub(startTime);
+    if (timeSinceInitialization < SECONDS_LINEAR) {
+      // Pay out half of all block rewards linearly.
+      uint256 linearRewards = GOLD_SUPPLY_CAP.sub(GENESIS_GOLD_SUPPLY).div(2);
+      uint256 targetRewards = linearRewards.mul(timeSinceInitialization).div(SECONDS_LINEAR);
+      return targetRewards.add(GENESIS_GOLD_SUPPLY);
+    } else {
+      // TODO(asa): Implement block reward calculation for years 15-30.
+      require(false);
+      return 0;
+    }
+  }
+
+  /**
+   * @notice Returns the rewards multiplier based on the current and target gold supplies.
+   * @param targetGoldSupplyIncrease The target increase in current gold supply.
+   * @return The rewards multiplier based on the current and target gold supplies.
+   */
+  function _getRewardsMultiplier(
+    uint256 targetGoldSupplyIncrease
+  )
+    internal
+    view
+    returns (FixidityLib.Fraction memory)
+  {
+    uint256 targetSupply = getTargetGoldTotalSupply();
+    uint256 totalSupply = getGoldToken().totalSupply();
+    uint256 remainingSupply = GOLD_SUPPLY_CAP.sub(totalSupply.add(targetGoldSupplyIncrease));
+    uint256 targetRemainingSupply = GOLD_SUPPLY_CAP.sub(targetSupply);
+    FixidityLib.Fraction memory ratio = FixidityLib.newFixed(remainingSupply).divide(
+      FixidityLib.newFixed(targetRemainingSupply)
+    );
+    if (ratio.gt(FixidityLib.fixed1())) {
+      FixidityLib.Fraction memory delta = ratio.subtract(FixidityLib.fixed1()).multiply(
+        rewardsMultiplierParams.adjustmentFactors.underspend
+      );
+      FixidityLib.Fraction memory r = FixidityLib.fixed1().add(delta);
```

Comment:
Wondering if these variables could get better names to make the logic more readable. E.g. `ratio` -> `remainingToTargetRatio` makes it clear in further computations what that variable is.